docs/page-objects/additional-requests.rst

```diff
@@ -4,799 +4,113 @@
-          :meth:`~.HttpResponse.css` and :meth:`~.HttpResponse.xpath` are
-          available.
-
-
-.. _httpclient:
-
-HttpClient
-==========
-
-The main interface for executing additional requests would be :class:`~.HttpClient`.
-It also has full support for :mod:`asyncio` enabling developers to perform
-additional requests asynchronously using :py:func:`asyncio.gather`,
-:py:func:`asyncio.wait`, etc. This means that :mod:`asyncio` could be used anywhere
-inside the Page Object, including the :meth:`~.ItemPage.to_item` method.
-
-In the previous section, we've explored how :class:`~.HttpRequest` is defined.
-Let's see a few quick examples to see how to execute additional requests using
-the :class:`~.HttpClient`.
-
-Executing a HttpRequest instance
---------------------------------
-
-.. code-block:: python
-
-    import attrs
-    import web_poet
-    from web_poet import validates_input
-
-
-    @attrs.define
-    class ProductPage(web_poet.WebPage):
-        http: web_poet.HttpClient
-
-        @validates_input
-        async def to_item(self):
-            item = {
-                "url": self.url,
-                "name": self.css("#main h3.name ::text").get(),
-                "product_id": self.css("#product ::attr(product-id)").get(),
-            }
-
-            # Simulate clicking on a button that says "View All Images"
-            request = web_poet.HttpRequest(f"https://api.example.com/v2/images?id={item['product_id']}")
-            response: web_poet.HttpResponse = await self.http.execute(request)
-
-            item["images"] = response.css(".product-images img::attr(src)").getall()
-            return item
-
-As the example suggests, we're performing an additional request that allows us
-to extract more images in a product page that might not be otherwise be possible.
-This is because in order to do so, an additional button needs to be clicked
-which fetches the complete set of product images via AJAX.
-
-There are a few things to take note of this example:
-
-    * Recall from the :ref:`httprequest-example` tutorial section that the
-      default method is ``GET``. Thus, the ``method`` parameter can be omitted
-      for simple ``GET`` requests.
-    * We're now using the ``async/await`` syntax inside the :meth:`~.ItemPage.to_item`
-      method.
-    * The response from the additional request is of type :class:`~.HttpResponse`.
-
-.. tip::
-
-    Check out the :ref:`http-batch-request-example` tutorial section to see how
-    to execute a group of :class:`~.HttpRequest` in batch.
-
-Fortunately, there are already some quick shortcuts on how to perform single
-additional requests using the :meth:`~.HttpClient.request`, :meth:`~.HttpClient.get`,
-and :meth:`~.HttpClient.post` methods of :class:`~.HttpClient`. These already
-define the :class:`~.HttpRequest` and executes it as well.
-
-.. _httpclient-get-example:
-
-A simple ``GET`` request
-------------------------
```

Comment:
I think that's a good call to remove most of the reference-like content from these docs. However, I think it could make sense to bring back some of the example. The don't need to be complete working examples with page objects (it'd be too much code), but it seems it'd be great to show

* http.get, http.post and other methods
* passing custom http headers
* some of the response features, like response.css, response.text and response.json()
* response.status and it's relationship with error handling

Something similar to https://requests.readthedocs.io/en/latest/user/quickstart/ probably, but without some sections?

Reply:
I think the goal here is for the users to be able to do 90% of their tasks without reading the reference docs.

Reply:
https://github.com/scrapinghub/web-poet/pull/197/commits/4f27bb879d1e9528e3fd6daf6d1ab8b6fedfd322